### PR TITLE
Filter out seen subjects from the subject queue

### PR DESCRIPTION
**Pull request by eatyourgreens**
Staging branch URL: https://seen-subjects.pfe-preview.zooniverse.org

Filters the incoming subject queue for any subjects that have been seen this session, but not yet marked as seen by Panoptes.

Also fixes some linter warnings for the classifier page:
- uses named class methods instead of anonymous functions.
- removes the unused `Translate` component.
- removes some semicolons and commas.

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by marten** [comment]
Is this going to be really annoying for project owners who're building a project when they've uploaded one or two subjects?

**Comment by eatyourgreens** [comment]
My test project only has one subject: https://seen-subjects.pfe-preview.zooniverse.org/projects/eatyourgreens/-project-testing-ground

**Comment by eatyourgreens** [comment]
The filter on seen and retired subjects has been in place since #3633, just over a year ago. I don't think we've seen any problems with API as a result of adding that filter.

File: app/pages/project/classify.jsx [hunk 1]
```diff
@@ -249,7 +248,12 @@ export class ProjectClassifyPage extends React.Component {
         }
       }).then((subjects) => {
         const nonLoadedSubjects = subjects.filter(newSubject => newSubject !== subjectToLoad);
-        const filteredSubjects = nonLoadedSubjects.filter(nonLoadedSubject => !nonLoadedSubject.already_seen && !nonLoadedSubject.retired);
+        const filteredSubjects = nonLoadedSubjects.filter((nonLoadedSubject) => {
+          const notSeen = !nonLoadedSubject.already_seen &&
+            !nonLoadedSubject.retired &&
+            !seenThisSession(workflow, nonLoadedSubject);
```
Comment (srallen) [inline app/pages/project/classify.jsx]: This is throwing a type error when I'm logged out. Looking over the function, I think this line needs to be `!seenThisSession.check(workflow, nonLoadedSubject)`

Comment (eatyourgreens) [reply app/pages/project/classify.jsx]: Thanks for catching this. It should be fixed now.

**Pull request merged**